### ISSUE TITLE
[Documentation]: Creating a new NWB file with copies of linked datasets requires new object IDs

### ISSUE THREAD

**Issue by celefthe** [NeurodataWithoutBorders/pynwb#1683]
### What would you like changed or added to the documentation and why?

Hey folks,

I tried exporting an NWB file with copies of linked datasets as described [here](https://pynwb.readthedocs.io/en/stable/export.html#my-nwb-file-contains-links-to-datasets-in-other-hdf5-files-how-do-i-create-a-new-nwb-file-with-copies-of-the-datasets). 

To cut a long story short, the example 
```python
with NWBHDF5IO(self.read_path, mode='r') as read_io:
    nwbfile = read_io.read()
    # nwbfile contains a TimeSeries where the TimeSeries data array is a link to an external dataset
    # in a different HDF5 file than self.read_path

    with NWBHDF5IO(self.export_path, mode='w') as export_io:
        export_io.export(src_io=read_io, nwbfile=nwbfile, write_args={'link_data': False})
```
resulted in a new NWB file with linked data rather than the data being copied over -- essentially a copy of the original NWB file.

 After some digging, I found that generating new object IDs prior to export fixed the issue, creating a new NWB file with copies of previously linked datasets.
```python
with NWBHDF5IO(self.read_path, mode='r') as read_io:
    nwbfile = read_io.read()
    nwbfile.generate_new_id()  # Looks like this is needed to export copies of externally linked datasets?

    with NWBHDF5IO(self.export_path, mode='w') as export_io:
        export_io.export(src_io=read_io, nwbfile=nwbfile, write_args={'link_data': False})
```

I'm not sure whether this was down to a peculiarity in the way I linked data to begin with (I don't think I've done anything particularly exotic), but without `nwbfile.generate_new_id()` the export function seemed to completely ignore the `link_data` parameter. 

I've not come across anybody reporting the same issue, but I wonder if it's worth updating the docs to include the line above just to catch whatever edge case I might have run into? I can't think of any reason why generating new object IDs would result in something terrible happening, but I'm probably not the best equipped to make that call. 

Thanks for all the work, NWB is good stuff 💯 

### Do you have any interest in helping write or edit the documentation?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

**Comment by celefthe** [comment]
Not sure whether this is a bug or (potentially) just a missing step in the docs; export otherwise works fine.

**Comment by bendichter** [comment]
Thanks for the report, @celefthe . I'm glad to see you found a work-around.

@rly I think you're probably the best person to answer this question

**Comment by rly** [comment]
Thanks for the report. I can confirm that this occurs. This is not intended. I created a minimum working example and bug report in HDMF where the issue most likely lies: https://github.com/hdmf-dev/hdmf/issues/842

**Comment by rly** [comment]
This has been fixed in HDMF and will be released in HDMF 3.5.5 this week.
